### ISSUE TITLE
assert triggered during propagation, potentially due to network error

### ISSUE THREAD

**Issue by noMICROSOFTbuhtz** [owncloud/client#7133]
I have no idea what is going on here. The server is not mine.
The "userdata-mlu" is a "fileserver" connected via SMB to my owncloud account.

What can I do? The client runs on Debian stable.
```
04-06 09:23:37:363 [ warning sync.propagator ]:	Could not complete propagation of "userdata-mlu/MyJabRef-BIB/nicht_in_JabRef_LIBRARY/Biometrie/Statistik_und_Co" by OCC::PropagateIgnoreJob(0x55a70feb8cf0) with status 6 and error: "Symbolische Verknüpfungen werden bei der Synchronisation nicht unterstützt."
04-06 09:23:37:363 [ info sync.propagator ]:	Starting INSTRUCTION_SYNC propagation of "userdata-mlu/MyJabRef-BIB/own-exported.bib" by OCC::PropagateUploadFileV1(0x55a70ff247d0)
04-06 09:23:37:363 [ info sync.checksums ]:	Computing "SHA1" checksum of "/home/user/ownCloud/userdata-mlu/MyJabRef-BIB/own-exported.bib" in a thread
04-06 09:23:37:363 [ info sync.propagator.upload ]:	"/userdata-mlu/MyJabRef-BIB/own-exported.bib" "SHA1:cec799668370e648196edc59a0944990ee109d9a"
04-06 09:23:37:363 [ info sync.accessmanager ]:	3 "" "https://cloud.uni-halle.de/remote.php/dav/files/akcpa/userdata-mlu/MyJabRef-BIB/own-exported.bib" has X-Request-ID "93cf3789-a708-4aab-9c80-3471626977c1"
04-06 09:23:37:363 [ info sync.networkjob ]:	OCC::PUTFileJob created for "https://cloud.uni-halle.de" + "/userdata-mlu/MyJabRef-BIB/own-exported.bib" "OCC::PropagateUploadFileV1"
04-06 09:23:38:078 [ info sync.networkjob.put ]:	PUT of "https://cloud.uni-halle.de/remote.php/dav/files/akcpa/userdata-mlu/MyJabRef-BIB/own-exported.bib" FINISHED WITH STATUS QNetworkReply::NetworkError(NoError) "" QVariant(int, 204) QVariant(QString, "No Content")
04-06 09:23:38:079 [ warning sync.propagator.upload ]:	Server does not support X-OC-MTime ""
04-06 09:23:38:079 [ info sync.database ]:	Setting blacklist entry for  "userdata-mlu/MyJabRef-BIB/own-exported.bib" 1 "Server does not support X-OC-MTime" 1554535418 0 1554457546 "5ca853f9ed116" "" 0
04-06 09:23:38:079 [ warning sync.propagator ]:	Could not complete propagation of "userdata-mlu/MyJabRef-BIB/own-exported.bib" by OCC::PropagateUploadFileV1(0x55a70ff247d0) with status 3 and error: "Server does not support X-OC-MTime"
04-06 09:23:38:080 [ info sync.database ]:	Updating file record for path: "userdata-mlu/MyJabRef-BIB/own-exported.bib" inode: 58721454 modtime: 1554457546 type: 0 etag: "5ca853f9ed116" fileId: "27006215ocd551ea6ca7" remotePerm: "WDNVRm" fileSize: 18492 checksum: "SHA1:cec799668370e648196edc59a0944990ee109d9a"
04-06 09:23:38:080 [ warning sync.propagator ]:	Could not complete propagation of "userdata-mlu/MyJabRef-BIB/own-exported.bib" by OCC::PropagateUploadFileV1(0x55a70ff247d0) with status 4 and error: "Server does not support X-OC-MTime"
04-06 09:23:38:081 [ critical default ]:	ASSERT: "i >= 0" in file /build/owncloud-client-004stS/owncloud-client-2.4.1+dfsg/src/libsync/owncloudpropagator.cpp, line 821
Speicherzugriffsfehler

```

**Comment by ckamm** [comment]
Thanks for the report! It's possible that there are two issues here: The ASSERT is definitely a client bug, it's possible it gets triggered by another issue relating to the SMB external storage on the server.

**Comment by ckamm** [comment]
@noMICROSOFTbuhtz You're using an old client, before 2.5.1. Which version was this exactly? Can you reproduce the issue? If so, please try with 2.5.4. The code handling "Server does not support X-OC-MTime" has been changed.

**Comment by ckamm** [comment]
This bug was likely fixed for 2.5.0 by 3f4d504c34778a392c1e4c91df6616101e0a9d9c see #6403.
